App/Http/Controllers/RoleController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\Eloquent\RoleRepository;
+use App\Enums\UserPermissionsEnum;
+use App\Models\Role;
+use App\Models\Permission;
+
+class RoleController extends Controller
+{
+
+
+    public function __construct()
```

Comment:
inject here the RoleRepository and use a variable to initialize it and replace the below places where it is used.

With this way, Laravel can do the IoC and find out from the service provider. 

Also I can see you didn't create a service provider to register the repositories.

You can create a RepositoryServiceProvider under App/Providers and insider register you can define this
```
....
        $this->app->singleton(RoleRepository::class, function ($app) {
            return new RoleRepository();
        });
```
and also add the method 

```
/**
     * Get the services provided by the provider.
     *
     * @return array
     */
    public function provides()
    {
        return [
            RoleRepository::class,
        ];
     }
```

When you come here and declare the RoleRepository you can do this

```
$this->roleRepository = app(RoleRepository::class);
```

This way you can change from your Service Provider the source of RoleRepository and fetch data lets say from Elasticsearch and code will never know as long as you keep the Interface and return types correct.

Apply this on all controllers.

Reply:
done